tap_salesforce/salesforce/__init__.py

```diff
@@ -143,8 +143,8 @@ def field_to_property_schema(field, mdata): # pylint:disable=too-many-branches
         property_schema['type'] = "string"
     elif sf_type in DATE_TYPES:
         date_type = {"type": "string", "format": "date-time"}
-        string_type = {"type": ["string", "null"]}
-        property_schema["anyOf"] = [date_type, string_type]
+        property_schema["format"] = "date-time"
+        property_schema["type"] = ["string", "null"]
```

Comment:
anyOf is not supported in the redshift target. Using anyOf leads to the redshift target creating multiple columns in redshift, one for the string type and one for the date time type.